modules/space/rendering/renderablesatellites.cpp

```diff
@@ -252,15 +247,25 @@ void RenderableSatellites::readDataFile(const std::string& filename) {
 
 void RenderableSatellites::initializeFileReading() {
     _startRenderIdx.removeOnChange(_startRenderIdxCallbackHandle);
-    _sizeRender.removeOnChange(_sizeRenderCallbackHandle);
     _startRenderIdx.setMaxValue(static_cast<unsigned int>(_numObjects - 1));
-    _sizeRender.setMaxValue(static_cast<unsigned int>(_numObjects));
-    _startRenderIdx = static_cast<unsigned int>(0);
-    _sizeRender = static_cast<unsigned int>(_numObjects);
+    if (_propsDefinedInAssetFlag.startRenderIdx) {
+        _propsDefinedInAssetFlag.startRenderIdx = false;
+    }
+    else {
+        _startRenderIdx = static_cast<unsigned int>(0);
+    }
     _startRenderIdxCallbackHandle = _startRenderIdx.onChange(
         _updateStartRenderIdxSelect);
-    _sizeRenderCallbackHandle = _sizeRender.onChange(
-        _updateRenderSizeSelect);
+
+    _sizeRender.removeOnChange(_sizeRenderCallbackHandle);
+    _sizeRender.setMaxValue(static_cast<unsigned int>(_numObjects));
+    if (_propsDefinedInAssetFlag.sizeRender) {
```

Comment:
I was thinking since the values are inialized to 0 or the dictionary value, maybe you dont need this _propsDefinedInAssetFlag thing and you could just check sizeRender and upperLimit against 0.

`if (_sizeRender == 0) _sizeRender = static_cast<unsigned int>(_numObjects);`

Reply:
I agree this would be cleaner. I put these flags in because I was having trouble with the order of execution; it was the only way to prevent it from overriding dictionary settings from a file.However, this was before I switched to not calling InitializeGL outside of the render call with each frame.